### PR TITLE
Added RPS command

### LINKED ISSUE #613: Rock-Paper-Scissors Command
## Description
Rock-paper-scissors is a very common game that can be really interesting to play, especially on Discord. It would be a good choice to have it on Sir LanceBot, known for its simple but fun and interesting games.

## Reasoning
This is a simple game that can instantly cure boredom and can also be very engaging at times. It would be used by many on the server and it also deserves to be added along with the other games, such as tic-tac-toe and battleship.

## Proposed Implementation
It is a pretty simple code and the outcome should be as shown in the image below. __The image is of course not the final output, I will improvise the commands and modify the results according to your needs and requirements.__

Here is what the command would be:
Rock-paper-scissors command: `rps` with an alias - `rockpaperscissors`

This command can be used without an argument but adding an argument will take the user's argument into consideration, also concluding who won in the end.

This command will simply be made using the `random` module.

image

## Would you like to implement this yourself?

- [x] I'd like to implement this feature myself
- [ ] Anyone can implement this feature

### PR DESCRIPTION
**Pull request by OculusMode**
Closes #613 

## Description
Added Rock paper scissors command which allows players to play it with bot.

## Format
.rps <option_name>

where option_name can be any of r, p, s, rock, scissor, paper in any case.
## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

**Review by Shivansh-007 — CHANGES_REQUESTED** [review]
Hello, and thanks for your contribution!

There are some lint issues with your PR, can you please fix those?

I'd also like you to try to write more descriptive commit messages. For example, rather than `minor changes` you can write `Apply grammar changes to invalid move message`, etc.

---

Since the code is relatively small, we can merge `Game` and the cog class together.

**Comment by OculusMode** [comment]
@Shivansh-007 Thanks a lot for reviewing I have made changes you requested and resolved all the flake8 errors the PR faced.
Just a small review of changes done:

- Merged Game class and cog class
- Added Doc strings in all the methods
- Capitalised static variables
- Added missing types
- Removed dead code
- Changed @commands.groups to @commands.command
- Changed variable name to appropriate meaningful name(i.e. arg => move)
- Checked linting on local machine and tested bot in my server

Thanks, please let me know if more changes are required.

File: bot/exts/evergreen/rps.py
Comment (OculusMode) [reply bot/exts/evergreen/rps.py:71]: Great, I'll wait for reply of core devs if its fine or not and will change if required.

File: bot/exts/evergreen/rps.py
Comment (OculusMode) [reply bot/exts/evergreen/rps.py]: Changed

File: bot/exts/evergreen/rps.py
Comment (OculusMode) [reply bot/exts/evergreen/rps.py]: removed helper command.

File: bot/exts/evergreen/rps.py
Comment (OculusMode) [reply bot/exts/evergreen/rps.py]: changed as requested.

**Review by Objectivitix — COMMENTED** [review]
It might be better to add some logical spacing here to facilitate reading :D
image

**Comment by OculusMode** [comment]
> It might be better to add some logical spacing here to facilitate reading :D
> image

@Objectivitix yeah that will give better readability indeed!
Great! I'll update it soon!

File: bot/exts/evergreen/rps.py
Comment (Shivansh-007) [inline bot/exts/evergreen/rps.py:73]: This is redundant.

**Review by Shivansh-007 — CHANGES_REQUESTED** [review]

File: bot/exts/evergreen/rps.py
Comment (OculusMode) [reply bot/exts/evergreen/rps.py]: Made required changes.

**Comment by OculusMode** [comment]
@Objectivitix made required changes.!

**Review by Shivansh-007 — APPROVED** [review]

File: bot/exts/evergreen/rps.py
Comment (Shivansh-007) [reply bot/exts/evergreen/rps.py:75]: cc @Xithrius @ChrisLovering

Comment (Xithrius) [reply bot/exts/evergreen/rps.py:75]: Feel free to remove this decorator.

Comment (OculusMode) [reply bot/exts/evergreen/rps.py]: @Xithrius removed guild_only and checked on local, seems good!

File: bot/exts/evergreen/rps.py
Comment (Xithrius) [inline bot/exts/evergreen/rps.py:44]: Given that these functions are used only once, I don't think there's any point to separating them from where they're being called from.

File: bot/exts/evergreen/rps.py
Comment (Xithrius) [inline bot/exts/evergreen/rps.py:8]: One newline will do.
```suggestion

```

File: bot/exts/evergreen/rps.py
Comment (Xithrius) [inline bot/exts/evergreen/rps.py:72]: Why not just have the contents of this function be within the command?

File: bot/exts/evergreen/rps.py
Comment (Xithrius) [inline bot/exts/evergreen/rps.py:13]:
```suggestion
# Using a dictionary instead of conditions to check for the winner.
```

**Review by Xithrius — CHANGES_REQUESTED** [review]

File: bot/exts/evergreen/rps.py
Comment (OculusMode) [reply bot/exts/evergreen/rps.py]: I have removed them in new commit. Thanks.

File: bot/exts/evergreen/rps.py
Comment (Xithrius) [inline bot/exts/evergreen/rps.py:58]:
```suggestion
        if player_result == 0:
            message_string = f"{player_mention} You and Sir Lancebot played {bot_move}, it's a tie."
            await channel.send(message_string)
        elif player_result == 1:
            await channel.send(f"Sir Lancebot played {bot_move}! {player_mention} won!")
        else:
            await channel.send(f"Sir Lancebot played {bot_move}! {player_mention} lost!")
```

**Review by Xithrius — CHANGES_REQUESTED** [review]
We should be using `await ctx.send()` instead of `await channel.send()` after setting `channel` to `ctx.channel`.

**Comment by OculusMode** [comment]
> We should be using `await ctx.send()` instead of `await channel.send()`.

Yeah I think we can do that as this is kinda 1 time command. Great, I'll change it asap and recheck.

**Review by anand2312 — COMMENTED** [review]
Pretty neat, I like the dict implementation for deciding the winners 👌🏻 
Just some minor comments

File: bot/exts/evergreen/rps.py
Comment (anand2312) [reply bot/exts/evergreen/rps.py]: Turns out this got resolved the second I commented this 😔

Comment (OculusMode) [reply bot/exts/evergreen/rps.py]: Haha, I heard it from future and changed it before you mention!

File: bot/exts/evergreen/rps.py
Comment (ToxicKidz) [inline bot/exts/evergreen/rps.py:41]: This should come out as an error
```suggestion
            raise commands.BadArgument(f"Invalid move. Please make move from options: {', '.join(CHOICES).upper()}.")
```

**Review by ToxicKidz — CHANGES_REQUESTED** [review]

**Review by ToxicKidz — CHANGES_REQUESTED** [review]
There are two things that should be changed:
1. Single quotes are being used at the top of the file, however, our [style guide](https://pythondiscord.com/pages/guides/pydis-guides/contributing/style-guide/#quote-marks) says that we should prefer double quotes over single quotes.
2. Change the spelling of `scissor` to `scissors`.

File: bot/exts/evergreen/rps.py
Comment (ToxicKidz) [inline bot/exts/evergreen/rps.py:35]:
```suggestion
        """Play the classic game of Rock Paper Scissors with your own sir-lancebot!"""
```

File: bot/exts/evergreen/rps.py
Comment (ToxicKidz) [inline bot/exts/evergreen/rps.py:31]:
```suggestion
    """Rock Paper Scissors. The Classic Game!"""
```

**Review by ToxicKidz — CHANGES_REQUESTED** [review]

**Review by ToxicKidz — APPROVED** [review]
:shipit:

**Review by kosayoda — APPROVED** [review]

**Comment by kosayoda** [comment]
Thanks for the PR!

**Pull request merged**
